### PR TITLE
feat(feedback): Improve the icon for Linked Errors in feedback details

### LINKED ISSUE #61368: Bug: Ticket icon shows on feedback list for feedbacks without tickets
image

While:
image

### PR DESCRIPTION
**Pull request by ryan953**
I made a few changes related to how we render the "Linked Error" (previously "Linked Issue") section:
- Icon and title changed
- If there is an id, but the data cannot be found, show that
- Show the section title while we're loading (we only start loading if we think there will be data)
- Extra: Remove extra ErrorBoundary components, and hoist everything up so it's consistent and easier to audit.

**Before:**

**Updated icon and "Linked Error" section:**
| desc | screen shot |
| --- | --- |
| List item with error | list item with error |
| List item with error and issue-tracker | list item github tooltip  list item error tooltip |
| Details loading | details loading |
| Details error not found | details not found |

Fixes https://github.com/getsentry/sentry/issues/61368

### CODE REVIEW

File: static/app/components/feedback/feedbackItem/useFetchCrashReport.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+export default function useFetchCrashReport({
+  crashReportId,
+  organization,
+  projectSlug,
+}: Props) {
+  const eventEndpoint = `/projects/${organization.slug}/${projectSlug}/events/${crashReportId}/`;
+  const {data: eventData, isFetching: isEventFetching} = useApiQuery<Event>(
+    [eventEndpoint],
+    {retryDelay: 100, staleTime: 0}
```
Comment (michellewzhang) [inline static/app/components/feedback/feedbackItem/useFetchCrashReport.tsx:18]: what's the reasoning for retrying?

**Review by michellewzhang — APPROVED** [review]
nice refactor! just curious, what is the reason we didn't need so many `ErrorBoundary` components/why did we have so many in the first place?

File: static/app/components/feedback/feedbackItem/useFetchCrashReport.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/components/feedback/feedbackItem/useFetchCrashReport.tsx:18]: good question! the default is so slow. 

I can add a comment about that.

**Comment by ryan953** [comment]
> nice refactor! just curious, what is the reason we didn't need so many `ErrorBoundary` components/why did we have so many in the first place?

@michellewzhang I think that some of the ErrorBoundary instances we from copy+paste. They're useful to catch errors, but buttons can't throw any errors so I thought we can get rid of the excess. I left in the ErrorBoundary calls around anything that fetches data, and moved them all up into the feedbackItem.tsx file

**Comment by billyvg** [comment]
> They're useful to catch errors, but buttons can't throw any errors so I thought we can get rid of the excess. I left in the ErrorBoundary calls around anything that fetches data, and moved them all up into the feedbackItem.tsx file

@michellewzhang ErrorBoundary are also good to have to minimize damage to the UI. Errors bubble up to the boundary where you render a minimal error state so that it can leave your UI still usable.

e.g. the KeyValueTable, if you had a single problematic row, the error boundary would show an error component for that single row, instead of the entire table

Agreed with the buttons though, it's not as helpful there.

**Pull request merged**